runtime/lua/vim/lsp/util.lua

```diff
@@ -684,19 +684,27 @@ do
     return severity_highlights[severity]
   end
 
-  function M.show_line_diagnostics()
+  function M.get_line_diagnostics()
     local bufnr = api.nvim_get_current_buf()
     local line = api.nvim_win_get_cursor(0)[1] - 1
+
+    local buffer_diagnostics = M.diagnostics_by_buf[bufnr]
+    if not buffer_diagnostics then
+      return {}
+    end
+
+    local line_diagnostics = buffer_diagnostics[line]
```

Comment:
I think this is now outdated as the `diagnostics_by_buf` are no longer grouped by line by default, and there is also https://github.com/neovim/neovim/pull/12244 which will change the format further.

Reply:
I have updated it to group by line first. I can change it again if that PR is merged before this one.